R/archive.R

```diff
@@ -377,6 +354,63 @@ new_epi_archive <- function(
   )
 }
 
+#' given a tibble as would be found in an epi_archive, remove duplicate entries.
+#' @description
+#' works by shifting all rows except the version, then comparing values to see
+#'   if they've changed. We need to arrange in descending order, but note that
+#'   we don't need to group, since at least one column other than version has
+#'   changed, and so is kept.
+#' @keywords internal
+#' @importFrom dplyr filter
+compactify_tibble <- function(df, keys) {
+  df %>%
+    arrange(!!!keys) %>%
+    filter(if_any(
+      c(everything(), -version), # all non-version columns
+      ~ !is_locf(.)
+    ))
+}
+
+#' get the entries that `compactify_tibble` would remove
+#' @keywords internal
+#' @importFrom dplyr filter if_all everything
+removed_by_compactify <- function(df, keys) {
+  df %>%
+    arrange(!!!keys) %>%
+    filter(if_all(
+      c(everything(), -version),
+      ~ is_locf(.)
+    )) # nolint: object_usage_linter
+}
+
+#' Checks to see if a value in a vector is LOCF
+#' @description
+#' LOCF meaning last observation carried forward. lags the vector by 1, then
+#'   compares with itself. For floats it uses float comparison, otherwise it
+#'   uses equality.
+#' `NA`'s are considered equal to `NA`'s, while nan's are not.
```

Comment:
either make this
```
#' `NA`s and `NaN`s are considered equal to each other.
```
or update behavior to be as described.

Reply:
not sure how I drew this conclusion incorrectly. Added a test for this (and the floating point equality).